disk.go

```diff
@@ -13,31 +13,48 @@ const (
 	DqdChunks
 )
 
+var idCounter int
 //==================================================================
 //
 // type: Disk
 //
 //==================================================================
+
+// DiskRunnerBase implementing DiskRunnerInterface
+// It is same as RunnerBase for now. More methods
+// will be added as the simulation evolves.
+type DiskRunnerBase struct {
+	RunnerBase
+}
+
 type Disk struct {
-	node       RunnerInterface
-	MBps       int
-	reserved   int
-	lastIOdone time.Time
-	writes     int64
-	reads      int64
-	writebytes int64
-	readbytes  int64
+	DiskRunnerBase
+	node           NodeRunnerInterface
+	MBps           int
+	reserved       int
+	lastIOdone     time.Time
+	writes         int64
+	reads          int64
+	writebytes     int64
+	readbytes      int64
 }
 
-func NewDisk(r RunnerInterface, mbps int) *Disk {
-	d := &Disk{node: r, MBps: mbps, reserved: 0, lastIOdone: Now}
+func NewDisk(r NodeRunnerInterface, mbps int) *Disk {
+	idCounter++
+	d := &Disk{
+		DiskRunnerBase: DiskRunnerBase{RunnerBase{id: idCounter}},
+		node: r, MBps: mbps, reserved: 0, lastIOdone: Now}
 	return d
 }
 
 func (d *Disk) String() string {
 	return fmt.Sprintf("disk-%s,%d,%d,%d,%d", d.node.String(), d.writes, d.reads, d.writebytes, d.readbytes)
 }
 
+// Dummy Run function to satisfy DiskRunnerInterface
+func (d *Disk) Run() {
+}
```

Comment:
Maybe better to divide disks into those that are runners and those that are not. This empty function sort of makes the case to do this.